webcomponents/tool/src/main/frontend/packages/sakai-rubrics/src/SakaiRubricsList.js

```diff
@@ -27,25 +43,41 @@ export class SakaiRubricsList extends RubricsElement {
 
   get siteId() { return this._siteId; }
 
-  search(search) {
-
-    this.querySelectorAll("sakai-rubric, sakai-rubric-readonly").forEach(rubric => {
+  search(term) {
+    this._searchTerm = term;
+    this._currentPage = 1;
+    this._lastCreatedRubricId = null;
+    this.requestUpdate();
+  }
 
-      rubric.classList.remove("d-none");
-      rubric.classList.toggle("d-none", !rubric.matches(search));
-    });
+  getFilteredRubrics() {
+    if (!this._rubrics) return [];
+    if (!this._searchTerm) return this._rubrics;
+    const term = this._searchTerm.toLowerCase();
+    return this._rubrics.filter(r =>
+      (r.title && r.title.toLowerCase().includes(term)) ||
+      (r.siteTitle && r.siteTitle.toLowerCase().includes(term)) ||
+      (r.creatorDisplayName && r.creatorDisplayName.toLowerCase().includes(term)) ||
+      (this._lastCreatedRubricId && r.id === this._lastCreatedRubricId)
+    );
   }
 
   shouldUpdate() {
     return this._rubrics && super.shouldUpdate();
   }
 
   render() {
+    const filteredRubrics = this.getFilteredRubrics();
```

Comment:
These computations should not be happening here. Having these here means that every time a reactive property changes, all these computations will occur. You should be able to move them into different parts of the class, like where you set this._currentPage.